packages/web-app-files/src/components/SideBar/Shares/Collaborators/CollaboratorsEditOptions.vue

```diff
@@ -335,6 +381,34 @@ export default {
       return role.name === 'advancedRole'
     },
 
+    confirmCustomPermissions() {
+      this.$refs.customPermissionsDrop.hide()
+
+      this.selectedRole = this.advancedRole
+      this.publishChange()
+    },
+
+    cancelCustomPermissions() {
+      this.$refs.customPermissionsDrop.hide()
+      this.$refs.rolesDrop.show()
+    },
+
+    getRecipientsCustomPermissions() {
+      // Custom permissions are only available for the custom role
+      // so do not display checked permissions if a user is switching from a different role
+      if (this.existingRole !== undefined && this.existingRole.name !== 'advancedRole') {
+        return
+      }
+
+      this.customPermissions = []
```

Comment:
This and the next few lines could be simplified to:
```js
this.customPermissions = this.collaboratorsPermissions
        ? [...this.collaboratorsPermissions]
        : []
```
or any reason against it?